### DISCUSSION TITLE
“绕过中国大陆 IP”如何自定义不绕过的域名后缀

### DISCUSSION THREAD

**Discussion by Yonick2017** [vernesong/OpenClash#2903] [Q&A]
各位大神，想问下“绕过中国大陆 IP”功能中，能否自定义不绕过（也就是要进入Clash代理）的域名后缀？
### 缘由
是这样的，我的openclash部署在旁路网关上，我希望开启绕过中国大陆IP这个功能来提升性能。然而在使用的时候，有一些**中国大陆**的域名以及IP需要通过socks5代理来进行访问。相关规则在配置文件里面里面已经列出来，比如说访问大学相关网站的时候匹配校园网代理规则。
```
- DOMAIN-KEYWORD,Tsinghua,TsinghuaNet
- DOMAIN-SUFFIX,chineseall.cn,TsinghuaNet
-  IP-CIDR,116.42.78.0/18,TsinghuaNet,no-resolve
```
由于这些域名以及IP都在中国大陆范围内，所以按理来说会被直接过滤掉，进行转发直连，连代理规则配置都不会进行匹配。所以想来问一问有没有方法可以排除掉**部分国内地址**让他进入Clash内核进行规则判断处理。
### 尝试过的解决方案
我看到OpenClash里面有一项设置是“绕过中国大陆 IPv4 黑名单”，用于“列表中的域名或 IP 将不会受到绕过中国大陆 IP 选项的影响，此功能依赖于 Dnsmasq”，并给出了以下例子：
```
#baidu.com
#114.114.114.114
```
但是我没有办法列举出符合条件的所有完整IP地址、所有完整域名，所以想问一问有没有方法可以实现像上面三行一样可以实现**域名关键词匹配**、**域名后缀匹配**以及**ip段匹配**？
我测试过在“绕过中国大陆 IPv4 黑名单”输入：
```
tsinghua.edu.cn
*.tsinghua.edu.cn
```
实际上访问 `cs.tsinghua.edu.cn` 的时候仍然不会被Clash代理。
请问这种情况有没有方法解决？谢谢！

**Comment by fxdw09** [comment]
我也遇到这个问题了，选择了绕过大陆IP，结果自定义了国内某地址经过代理设置后，始终不生效；关闭绕过大陆IP后，生效了，求助

**Reply by 2thcmoon** [reply]
同样遇到这个问题
